docs/decisions/0023-planner-telemetry-enhancement.md

```diff
@@ -0,0 +1,218 @@
+---
+status: { proposed }
+contact: { TaoChenOSU }
+date: { 2023-11-21 }
+deciders: alliscode, dmytrostruk, markwallace, SergeyMenshykh, stephentoub
+consulted: {}
+informed: {}
+---
+
+# Planner Telemetry Enhancement
+
+## Context and Problem Statement
+
+It would be extremely beneficial for applications using Semantic Kernel' planning features to be able to continuously monitor the performance of planners and plans.
+
+## Scenarios
+
+Contoso is a company that is developing an AI application using SK.
+
+1. Contoso needs to continuously monitor the token usage of a particular planner, including prompt tokens, completion tokens, and the total tokens.
+2. Contoso needs to continuously monitor the time it takes for a particular planner to create a plan.
+3. Contoso needs to continuously monitor the success rate of a particular planner in creating a valid plan.
+4. Contoso needs to continuously monitor the success rate of a particular plan type being executed successfully.
+5. Contoso wants to be able to see the token usage of a particular planner run.
+6. Contoso wants to be able to see the time taken to create a plan of a particular planner run.
+7. Contoso wants to be able to see the steps in a plan.
```

Comment:
```suggestion
7. Contoso wants to be able to see the steps and functions used in a plan.
```

Reply:
What are the differences between steps and functions?